### PR TITLE
[WIP, do not merge] Add solana wallets supported by portal bridge

### LINKED ISSUE #378: [enhancement] hide wallets that could not be used
For example, if an integration doesn't use Solana we should not show solana wallets in the modal. The same would be true for other network/context specific wallets.

### PR DESCRIPTION
**Pull request by M-Picco**
Also, use icons supplied by the wallet aggregator sdk wallets.

Also closes #378 

image

### CODE REVIEW

File: wormhole-connect/src/icons/WalletIcons.tsx [hunk 1]
```diff
@@ -16,241 +17,23 @@ const useStyles = makeStyles<{ size: number }>()((theme, { size }) => ({
...
-    }
-    default: {
-      return <div style={{ height: `${size}px`, width: `${size}px` }}></div>;
-    }
-  }
+  const { wallet } = props;
+
+  return wallet ? (
+    <div className={classes.container}>
+      <img className={classes.icon} src={wallet.getIcon()} alt={wallet.getName()} />
```
Comment (anondev2323) [inline wormhole-connect/src/icons/WalletIcons.tsx]: Could actually just do:
```
return (
  <div className={classes.container}>
    {wallet && <img className={classes.icon} src={wallet.getIcon()} alt={wallet.getName()} />}
  </div>
)
```

File: wormhole-connect/src/views/WalletModal.tsx [hunk 2]
```diff
@@ -62,40 +62,28 @@ type WalletData = {
...
-    isReady: getReady(wallets.solana.solflare),
-  },
+
+const mapWallets = (wallets: Record<string, Wallet>, type: WalletType): WalletData[] => {
+  return Object.values(wallets).map((wallet) => ({
+    name: wallet.getName(),
+    wallet,
+    type,
+    isReady: getReady(wallet),
+  }));
```
Comment (anondev2323) [inline wormhole-connect/src/views/WalletModal.tsx:72]: Nice, much cleaner/easy to maintain

**Comment by anondev2323** [comment]
#378

File: wormhole-connect/src/icons/WalletIcons.tsx (hunk 1 above)
Comment (M-Picco) [reply wormhole-connect/src/icons/WalletIcons.tsx]: I noticed the component is always used with a wallet prop, so no need to consider this case inside this component I think

**Comment by anondev2323** [comment]
Screen Shot 2023-05-02 at 2 50 18 PM

**Review by anondev2323 — APPROVED** [review]

**Comment by anondev2323** [comment]
Changes merged in #562

**Pull request closed without merging**